sys/include/net/ng_ipv6/addr.h

```diff
@@ -0,0 +1,482 @@
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.5.2">
+ *          RFC 4291, section 2.5.2
+ *      </a>
+ *
+ * @param[in] addr  An IPv6 address.
+ *
+ * @return  true, if @p addr is unspecified address
+ * @return  false, otherwise.
+ */
+static inline bool ng_ipv6_addr_is_unspecified(const ng_ipv6_addr_t *addr)
+{
+    return (addr->u64[0].u64 == 0) &&
+           (addr->u64[1].u64 == 0);
+}
+
+/**
+ * @brief   Check if @p addr is a multicast address.
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.7">
+ *          RFC 4291, section 2.7
+ *      </a>
+ *
+ * @param[in] addr  An IPv6 address.
+ *
+ * @return  true, if @p addr is multicast address,
+ * @return  false, otherwise.
+ */
+static inline bool ng_ipv6_addr_is_multicast(const ng_ipv6_addr_t *addr)
+{
+    return (addr->u8[0] == 0xff);
+}
+
+/**
+ * @brief   Checks if @p addr is a loopback address.
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.5.3">
+ *          RFC 4291, section 2.5.3
+ *      </a>
+ *
+ * @param[in] addr  An IPv6 address.
+ *
+ * @return  true, if @p addr is loopback address,
+ * @return  false, otherwise.
+ */
+static inline bool ng_ipv6_addr_is_loopback(const ng_ipv6_addr_t *addr)
+{
+    return addr->u64[0].u64 == 0 &&
+           byteorder_ntohll(addr->u64[1]) == 1;
+}
+
+/**
+ * @brief   Check if @p addr is a link-local address.
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.5.6">
+ *          RFC 4291, section 2.5.6
+ *      </a>
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.7">
+ *          RFC 4291, section 2.7
+ *      </a>
+ *
+ * @param[in] addr  An IPv6 address.
+ *
+ * @return  true, if @p addr is link-local address,
+ * @return  false, otherwise.
+ */
+static inline bool ng_ipv6_addr_is_link_local(const ng_ipv6_addr_t *addr)
+{
+    return (byteorder_ntohll(addr->u64[0]) == 0xfe80000000000000) ||
+           (ng_ipv6_addr_is_multicast(addr) &&
+            (addr->u8[1] & 0x0f) == NG_IPV6_ADDR_MCAST_SCP_LINK_LOCAL);
+}
+
+/**
+ * @brief   Check if @p addr is unique local unicast address.
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc4193">
+ *          RFC 4193
+ *      </a>
+ *
+ * @param[in] addr  An IPv6 address.
+ *
+ * @return  true, if @p addr is unique local unicast address,
+ * @return  false, otherwise.
+ */
+static inline bool ng_ipv6_addr_is_unique_local_unicast(const ng_ipv6_addr_t *addr)
+{
+    return ((addr->u8[0] == 0xfc) || (addr->u8[0] == 0xfd));
+}
+
+/**
+ * @brief   Check if @p addr is solicited-node multicast address.
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.7.1">
+ *          RFC 4291, section 2.7.1
+ *      </a>
+ *
+ * @param[in] addr  An IPv6 address.
+ *
+ * @return  true, if @p addr is solicited-node multicast address,
+ * @return  false, otherwise.
+ */
+static inline bool ng_ipv6_addr_is_solicited_node(const ng_ipv6_addr_t *addr)
+{
+    return (byteorder_ntohll(addr->u64[0]) == 0xff02000000000000) &&
```

Comment:
(also applies for https://github.com/RIOT-OS/RIOT/pull/2433#discussion_r24830428) I don't quite know what other use-case would be there for such a - then - public, other than using it here. The macros basically would be something like `NG_IPV6_ADDR_LINK_LOCAL_PREFIX_UINT64`. Then why not have a `NG_IPV6_ADDR_LINK_LOCAL_PREFIX_UINT32`, `NG_IPV6_ADDR_LINK_LOCAL_PREFIX_UINT16`, or `NG_IPV6_ADDR_LINK_LOCAL_PREFIX_UINT8`? Additionally for solicited node addresses the multicast address is prefixed with `0xff02000000000000` in this special case of the solicited node address. Link-local multicast addresses are generally prefixed with just `ff02::/4`, which would mean either more comparison or more weird public macros.

TL;DR: Too many special cases to provide them publically.